processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -808,6 +810,55 @@ public Object get()
                     }
 
                     @Override
+                    public NumericColumnSelector makeMathExpressionSelector(String expression)
+                    {
+                      final Expr parsed = Parser.parse(expression);
+                      final Set<String> required = Sets.newHashSet(Parser.findRequiredBindings(parsed));
+
+                      final Map<String, Supplier<Number>> values = Maps.newHashMapWithExpectedSize(required.size());
+                      for (String columnName : index.getColumnNames()) {
+                        if (!required.contains(columnName)) {
+                          continue;
+                        }
+                        final GenericColumn column = index.getColumn(columnName).getGenericColumn();
+                        if (column == null) {
+                          continue;
```

Comment:
It'll throw NPE in evaluation. MathExpressionSelector should be rewritten with following PR #2836, anyway